esp-hal/src/rsa/esp32sX.rs

```diff
@@ -120,7 +120,7 @@ impl<'d, DM: crate::Mode> Rsa<'d, DM> {
     }
 
     /// Clears the RSA interrupt flag.
-    pub(super) fn clear_interrupt(&mut self) {
+    pub(super) fn clear_interrupt(&self) {
```

Comment:
Out of curiosity why'd you remove the `mut` here?

Reply:
Because I was wrestling with this problem at midnight after 13 hours of work. There is probably an unreasonable amount of left-over from a bunch of back-and-forth changes. But the original reason was that the future holds on to `&Rsa` while at the same time we're doing operations on the same `&Rsa` so I needed this to be `&self` at one point. Probably not anymore, though.